staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -189,42 +211,56 @@ func (cfgCtlr *configController) initializeConfigController(informerFactory kube
 		AddFunc: func(obj interface{}) {
 			pl := obj.(*flowcontrol.PriorityLevelConfiguration)
 			klog.V(7).Infof("Triggered API priority and fairness config reloading due to creation of PLC %s", pl.Name)
-			cfgCtlr.configQueue.Add(0)
+			cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, obj)
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			newPL := newObj.(*flowcontrol.PriorityLevelConfiguration)
 			oldPL := oldObj.(*flowcontrol.PriorityLevelConfiguration)
 			if !apiequality.Semantic.DeepEqual(oldPL.Spec, newPL.Spec) {
 				klog.V(7).Infof("Triggered API priority and fairness config reloading due to spec update of PLC %s", newPL.Name)
-				cfgCtlr.configQueue.Add(0)
+				cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, newObj)
 			}
 		},
 		DeleteFunc: func(obj interface{}) {
 			name, _ := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 			klog.V(7).Infof("Triggered API priority and fairness config reloading due to deletion of PLC %s", name)
-			cfgCtlr.configQueue.Add(0)
+			cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, obj)
 
 		}})
 	fsi.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			fs := obj.(*flowcontrol.FlowSchema)
 			klog.V(7).Infof("Triggered API priority and fairness config reloading due to creation of FS %s", fs.Name)
-			cfgCtlr.configQueue.Add(0)
+			cfgCtlr.finishHandlingNotification(cfgCtlr.configQueue, obj)
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			newFS := newObj.(*flowcontrol.FlowSchema)
 			oldFS := oldObj.(*flowcontrol.FlowSchema)
-			if !apiequality.Semantic.DeepEqual(oldFS.Spec, newFS.Spec) {
-				klog.V(7).Infof("Triggered API priority and fairness config reloading due to spec update of FS %s", newFS.Name)
-				cfgCtlr.configQueue.Add(0)
+			if !(apiequality.Semantic.DeepEqual(oldFS.Spec, newFS.Spec) &&
+				apiequality.Semantic.DeepEqual(oldFS.Status, newFS.Status)) {
```

Comment:
It might be useful to add a comment explaining why we also want to compare status here.
You briefly mentioned it during one of the meetings, but I bet I will forget the explanation in the next 2 weeks :)

Reply:
done

Reply:
Sorry for not being clear - I meant to add a more explicit comment why we also want to recompute on Status changes.
In majority of controllers we only react for Spec.

Reply:
done

Reply:
Thanks - that's exactly what I wanted to see.